sql/src/main/java/org/apache/druid/sql/avatica/DruidConnection.java

```diff
@@ -63,7 +63,7 @@ public DruidConnection(final String connectionId, final int maxStatements, final
     this.connectionId = Preconditions.checkNotNull(connectionId);
     this.maxStatements = maxStatements;
     this.context = ImmutableMap.copyOf(context);
-    this.statements = new HashMap<>();
+    this.statements = new ConcurrentHashMap<>();
```

Comment:
Maybe we should remove the `GuardedBy` annotation from the `statements` instance and consider removing all `synchronized (statements) {...}` clauses since we use `ConcurrentHashMap` instead of `HashMap`

Reply:
We still need the createStatement logic to be synchronized to ensure the validity of checks like this one:

```java
      if (statements.size() >= maxStatements) {
        throw new ISE("Too many open statements, limit is[%,d]", maxStatements);
      }
```

Without synchronization, that won't actually verify maxStatements properly.

Reply:
Hi @asdf2014 , synchronized protected both statements and code blocked

Reply:
@gianm I see what you mean, Thanks!